akka-http-tests/src/test/scala/akka/http/scaladsl/marshallers/sprayjson/JsonStreamingSpec.scala

```diff
@@ -0,0 +1,170 @@
+    """.stripMargin
+
+  val arrayPretty =
+    """|[
+       |{"fname":"Frank","name":"Smith","age":42,"id":1337,"boardMember":false}
+       |{"fname":"Bob","name":"Smith","age":42,"id":1337,"boardMember":false}
+       |{"fname":"Hank","name":"Smith","age":42,"id":1337,"boardMember":false}
+       |]""".stripMargin
+
+  "JsonStreaming" should {
+    "read using entity(stream[T])" in {
+      val route = post {
+        entity(stream[Employee]) { employees ⇒
+          complete(employees.map(_.fname).intersperse(",").runFold("")(_ + _))
+        }
+      }
+
+      PostJson("/", lineByLineCommaSeparated) ~> route ~> check {
+        responseAs[String] shouldEqual "Frank,Bob,Hank"
+      }
+    }
+
+    "read entity(stream[JsObject])" in {
+      val route = post {
+        entity(stream[JsObject]) { js ⇒
+          complete(js.runFold("")({ case (acc, x) ⇒ acc + "." }))
+        }
+      }
+
+      PostJson("/", lineByLineCommaSeparated) ~> route ~> check {
+        status shouldEqual StatusCodes.OK
+      }
+    }
+
+    "reject when content type is not JSON" in {
+      val route = post {
+        entity(stream[Employee]) { employees ⇒
+          complete(employees.map(_.fname).intersperse(",").runFold("")(_ + _))
+        }
+      }
+
+      Post("/").withEntity(ContentTypes.`text/plain`, "Hello world!") ~> route ~> check {
+        handled shouldEqual false
+        rejection shouldEqual UnsupportedRequestContentTypeRejection(Set(ContentTypes.`application/json`))
+      }
+    }
+
+    "read using explicit framing" in {
+      val route = post {
+        entity(stream[Employee](framing = JsonFraming)) { employees ⇒
+          complete(employees.map(_.fname).intersperse(",").runFold("")(_ + _))
+        }
+      }
+
+      PostJson("/", lineByLineCommaSeparated) ~> route ~> check {
+        responseAs[String] shouldEqual "Frank,Bob,Hank"
+      }
+    }
+
+    "read in parallel (unordered) explicit framing" in {
+      val route = post {
+        entity(streamAsyncUnordered[Employee](parallelism = 3)) { employees ⇒
+          complete(employees.map(_.fname).intersperse(",").runFold("")(_ + _))
+        }
+      }
+
+      PostJson("/", lineByLineCommaSeparated) ~> route ~> check {
+        responseAs[String] shouldEqual "Frank,Bob,Hank"
+      }
+    }
+
+    "write streamed json line by line" in {
+      val frank = Employee.simple
+
+      val route = get {
+        complete(Source.repeat(frank).take(3))
+      }
+
+      val json =
+        """{"name":"Smith","boardMember":false,"fname":"Frank","age":42,"id":12345}""" + "\n" +
+          """{"name":"Smith","boardMember":false,"fname":"Frank","age":42,"id":12345}""" + "\n" +
+          """{"name":"Smith","boardMember":false,"fname":"Frank","age":42,"id":12345}""" // TODO what if someone wants eager newline?
+
+      Get("/") ~> route ~> check {
+        responseAs[String] shouldEqual json
+        contentType shouldEqual ContentTypes.`application/json`
```

Comment:
I agree with Mathias, I wouldn't use a non-standard extension of a common standard mime type because it's too easy to break something. (E.g. would it return something for client sending an `Accept: application/json` header?) 

Potential consequences would be less severe for a lesser used mime type like `application/x-ldjson`, so that's what I would prefer. Additionally, it would probably make sense to keep the media type easily configurable.